frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/TicfCriHandlerTest.java

```diff
@@ -0,0 +1,159 @@
+            throws IOException, InterruptedException, ExecutionException {
+        var ticfRequest = basicTicfCriRequest(COMMON_SUBJECTID, VTRS, JOURNEY_ID);
+        var expectedRequestBody =
+                format(
+                        "{\"sub\":\"%s\",\"vtr\":%s,\"govuk_signin_journey_id\":\"%s\",\"authenticated\":\"Y\"}",
+                        COMMON_SUBJECTID, jsonArrayFrom(VTRS), JOURNEY_ID);
+
+        when(httpResponse.statusCode()).thenReturn(200);
+        when(httpClient.send(any(), any())).thenReturn(httpResponse);
+        handler.handleRequest(ticfRequest, context);
+
+        var httpRequestCaptor = ArgumentCaptor.forClass(HttpRequest.class);
+        verify(httpClient).send(httpRequestCaptor.capture(), ArgumentMatchers.any());
+
+        var actualRequestBody =
+                bodyPublisherToString(httpRequestCaptor.getValue().bodyPublisher().get());
+
+        assertEquals(SERVICE_URI, httpRequestCaptor.getValue().uri());
+        assertEquals(expectedRequestBody, actualRequestBody);
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {200, 404, 500})
+    void sendsMetricsBasedOnTheHttpStatusCode(Integer statusCode)
+            throws IOException, InterruptedException {
+        var ticfRequest = basicTicfCriRequest(COMMON_SUBJECTID, VTRS, JOURNEY_ID);
+        var httpResponse = mock(HttpResponse.class);
+        when(httpResponse.statusCode()).thenReturn(statusCode);
+        when(httpClient.send(any(), any())).thenReturn(httpResponse);
+
+        handler.handleRequest(ticfRequest, context);
+
+        verify(cloudwatchMetricsService)
+                .incrementCounter(
+                        "TicfCriResponseReceived",
+                        Map.ofEntries(
+                                Map.entry("Environment", "test-environment"),
+                                Map.entry("StatusCode", statusCode.toString())));
+    }
+
+    @Test
+    void testIncrementsMetricWhenARequestTimesOut() throws Exception {
+        when(httpClient.send(any(), any()))
+                .thenThrow(new HttpConnectTimeoutException("request timed out"));
+
+        handler.handleRequest(basicTicfCriRequest(COMMON_SUBJECTID, VTRS, JOURNEY_ID), context);
+
+        verify(cloudwatchMetricsService).incrementCounter("TicfCriServiceTimeout", METRICS_CONTEXT);
+    }
+
+    @Test
+    void testIncrementsMetricWhenAnExceptionOccurs() throws Exception {
+        when(httpClient.send(any(), any())).thenThrow(new IOException("an IO Exception"));
+
+        handler.handleRequest(basicTicfCriRequest(COMMON_SUBJECTID, VTRS, JOURNEY_ID), context);
+
+        verify(cloudwatchMetricsService).incrementCounter("TicfCriServiceError", METRICS_CONTEXT);
+    }
+
+    private JsonArray jsonArrayFrom(List<String> elements) {
+        var jsonArray = new JsonArray();
+        elements.forEach(jsonArray::add);
+        return jsonArray;
+    }
+
+    private String bodyPublisherToString(HttpRequest.BodyPublisher bodyPublisher)
```

Comment:
This is somewhat ugly, but was the only way I could work out to parse the http request being made. If anyone can see a better way of doing this then I'd be very happy for suggestions :-D but otherwise I'd be pro getting this in and iterating.

Reply:
Agreed.